lib/lib_install.php

```diff
@@ -75,6 +75,101 @@ function checkRequirements(string $dbType = ''): array {
 	);
 }
 
+function getProcessUsername(): string {
+	if (function_exists('posix_getpwuid') && function_exists('posix_geteuid')) {
+		$processUser = posix_getpwuid(posix_geteuid()) ?: [];
+		if (!empty($processUser['name'])) {
+			return $processUser['name'];
+		}
+	}
+
+	if (function_exists('exec')) {
+		exec('whoami', $output);
+		if (!empty($output[0])) {
+			return $output[0];
+		}
+	}
+
+	return _t('install.check.unknown_process_username');
+}
+
+/** @return array<string> */
+function getOwnerOfFile(String $file): array {
+	clearstatcache();
+	$stat_file = @stat($file);
+	$owner = [];
+	if(!$stat_file) {
+		//Couldnt stat file
+		$owner['fileowner'] = 'unknown';
+		$owner['filegroup'] = 'unknown';
+		return $owner;
+	}
+	if (function_exists('posix_getpwuid') && function_exists('posix_getgrgid')) {
+		$owner['fileowner'] = @posix_getpwuid($stat_file['uid'])['name'];
+		$owner['filegroup'] = @posix_getgrgid($stat_file['gid'])['name'];
```

Comment:
I need help here. I cannot solf this PHPstan level 7 error:
```
 Line   lib/lib_install.php                                                 
 ------ -------------------------------------------------------------------- 
  109    Cannot access offset 'name' on array|false.                         
  110    Cannot access offset 'name' on array{name: string, passwd: string,  
         gid: int, members: array<int, string>}|false.      
```

Reply:
It is because `posix_getpwuid()` may return false.
If possible, try also on Windows as those functions are platform-dependent.

Reply:
fixed

Reply:
Windows has no user permissions things as far as I know. I tried the install process with XAMP on Win10:
![grafik](https://user-images.githubusercontent.com/1645099/234947593-d2c839a6-250f-4f75-9897-09223508f120.png)

Reply:
I meant that the such system functions might not exist / work on other platforms such as Windows (which does have user and group permissions - cf. NTFS rights), so we should make sure that it does not fail / crash on non-Linux platforms.